packages/p2p-v2/src/hooks/useSendbird.ts

```diff
@@ -0,0 +1,252 @@
+import { useCallback, useEffect, useRef, useState } from 'react';
+import { useAdvertiserInfo, useChatCreate, useOrderInfo, useSendbirdServiceToken, useServerTime } from '@deriv/api';
+import SendbirdChat, { BaseChannel, User } from '@sendbird/chat';
+import { GroupChannel, GroupChannelHandler, GroupChannelModule } from '@sendbird/chat/groupChannel';
+import { BaseMessage, MessageType, MessageTypeFilter, UserMessage } from '@sendbird/chat/message';
+
+const ChatMessageStatus = {
+    ERRORED: 'ERRORED',
+    PENDING: 'PENDING',
+} as const;
+
+type ChatMessage = {
+    channelUrl: string;
+    createdAt: number;
+    customType?: string;
+    fileType?: 'file' | 'image' | 'pdf';
+    id: string;
+    message?: string;
+    messageType: string;
+    name?: string;
+    senderUserId: string;
+    size?: number;
+    status?: keyof typeof ChatMessageStatus;
+    url?: string;
+};
+
+const getMessageType = (message: BaseMessage) => {
+    const isImageType = (type: string) => ['image/jpeg', 'image/png', 'image/gif'].includes(type);
+    const isPDFType = (type: string) => type === 'application/pdf';
+
+    if (message.isFileMessage()) {
+        if (isImageType(message.type)) {
+            return 'image';
+        } else if (isPDFType(message.type)) {
+            return 'pdf';
+        }
+        return 'file';
+    }
+};
+
+function createChatMessage(sendbirdMessage: BaseMessage): ChatMessage {
+    return {
+        channelUrl: sendbirdMessage.channelUrl,
+        createdAt: sendbirdMessage.createdAt,
+        customType: sendbirdMessage.customType,
+        fileType: getMessageType(sendbirdMessage),
+        id: sendbirdMessage.messageId.toString(),
+        message: sendbirdMessage.isUserMessage() ? sendbirdMessage.message : undefined,
+        messageType: sendbirdMessage.messageType,
+        name: sendbirdMessage.isFileMessage() ? sendbirdMessage.name : undefined,
+        senderUserId: (sendbirdMessage as UserMessage)?.sender.userId,
```

Comment:
@adrienne-deriv, wouldn't it be better to do this instead?
```suggestion
        senderUserId: (sendbirdMessage.isUserMessage() ? sendbirdMessage : undefined)?.sender?.userId ?? '',
```